### ISSUE TITLE
To be developed: Snapshot Functions

### ISSUE THREAD

**Issue by LanHikari22** [LanHikari22/C-ROM-Linker#8]
They take the form:
void snap<NUM_SNAPS>_<ABS_ROM_ADDR>();
An example:
void snap10_08040000();

What this does, is that it is going to  intercept at 0x08040000 (0x40000 in ROM), 10 times, for 10 different assembly instructions.
When intercepting, the logic at that instruction has to be modified to include branching logic. This must be accounted for to determine where the next instruction is, so that another intercept  occurs there.

Basically, for every one instruction run in a function, it will be intercepted by the logic in the snapshot function. This can be useful to view a (almost complete) state view of the intercepted function.  This can be very useful for diagnostics.
